### ISSUE TITLE
Get rid of "An explicit conversion exists (are you missing a cast?)" from CS0266

### ISSUE THREAD

**Issue by YairHalberstadt** [dotnet/roslyn#48431]
CS0266 has the following error message:

> Cannot implicitly convert type 'type1' to 'type2'. An explicit conversion exists (are you missing a cast?)

It comes up pretty much everywhere where the target type doesn't match the expression type, since almost any type can be explicitly converted to any other type. However in 99% of cases, it's not the correct solution, and so is just noise - it can also be confusing for users, who may add an explicit cast (thinking the diagnostic means an explicit cast operator exists) only to find things blow up at runtime.

I would suggest either getting rid of this diagnostic completely, or using `CS0029` "Cannot implicitly convert type 'int' to 'string'" in most cases, and adding new diagnostics for the following cases:

1. The target type is a subtype of the expression type (may want to downcast).
2. There is an explicit cast operator from the expression type to the target type (may want to use this cast).

**Comment by Joe4evr** [comment]
+100. The amount of times I've seen people doing something like `(User)o.Users` because they think that'll get them a single thing out of a collection of things (and proceed to complain that their code blows up) is staggering.

**Comment by CyrusNajmabadi** [comment]
I strongly agree wiht this.  The benefit of the proposal as well is that we can have much better diagnostics and diagnostic-ids for the cases that can be likely fixed up properly.  This can then better drive IDE features to do things like `insert the appropriate subtype cast`.  Tagging @jcouv to see if this is something that could be picked up.

**Comment by jcouv** [comment]
@YairHalberstadt Can you provide some scenario(s) to illustrate current (misleading) behavior with CS0266?
For `int` to `string`, I get "error CS0029: Cannot implicitly convert type 'string' to 'int'" (which is appropriate), not CS0266...

**Comment by YairHalberstadt** [comment]
@jcouv you get it when casting pretty much any non sealed reference type to another non sealed reference type (including interfaces)
